I3uckwheat/cards-of-carousal
# Fullscreen

#### 0. Make sure your issue is linked:



#### 1. Purpose:


Fullscreens the app for host and players

#### 2. Implementation:


Utilizes browser `requestFullscreen()` API.

#### 3. Possible Issues:


Needs testing on multiple browsers and devices.

#### 4. How to Test:



1. Edit `start.js` in the backend folder. The `app.listen` method needs a second argument, a string of '0.0.0.0' so the app is available on your local network:
```js
app.listen(process.env.EXPRESS_PORT, '0.0.0.0', () =>
  // eslint-disable-next-line no-console
  console.log(
    `Express listening at http://localhost:${process.env.EXPRESS_PORT}`,
  ),
);
```
2. Ensure all devices are on your local home network
3. Lookup the internal IP address of the computer running the app and write it down.
4. In frontend.env file, replace `localhost` with the IP address you just wrote down.
5. run `npm run dev:frontend` and `npm run dev:backend` in separate terminals 
6. On the host computer, you can use localhost just fine for the app, but on devices you are testing with, open the browser and navigate to `http://<HOST_COMPUTER_IP>:3000` (inside the angle brackets, replace the the text with the ip you wrote down earlier, and remove the angle brackets).
7. Host the game on one computer, join on any devices you want to test.
8. Click JOIN or HOST, and see if the game enters fullscreen mode.

#### 5. Cleanup
